From f37453f753d5a76006218cae4e2b31af019124fb Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 18 Sep 2015 15:31:18 +0530
Subject: [PATCH] fusb300: enable power for current measurement logic

when otg(mouse) is removed from S3 is not detected
due to the power not enabled to current measurement
circuit of fusb300 when going to suspend.

Change-Id: Id7be20759c42e79bb2659cb78fed23d5e69457f7
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-16974
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
---
 drivers/usb/typec/phy-fusb300.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 4e9adb6728d1..d72f06abc3b7 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -1806,7 +1806,8 @@ static int fusb300_late_suspend(struct device *dev)
 		phy->state == TYPEC_STATE_ATTACHED_DFP) {
 		/* enable power for wakeup block and measure block*/
 		regmap_write(chip->map, FUSB300_PWR_REG,
-			FUSB300_PWR_BG_WKUP | FUSB300_PWR_MEAS);
+			FUSB300_PWR_BG_WKUP | FUSB300_PWR_BMC |
+			FUSB300_PWR_MEAS);
 	} else {
 		/* enable power only for wakeup block */
 		regmap_write(chip->map, FUSB300_PWR_REG,
-- 
2.6.0.rc0.2.g7662973

